core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -198,6 +198,20 @@ UNWIND(	.cantunwind)
 UNWIND(	.fnend)
 END_FUNC reset
 
+	/* Setup SP to point to the top of the tmp stack for the current CPU */
```

Comment:
This comment from the _a64 file helps understanding the assembly:
*sp is assigned stack_tmp + (cpu_id + 1) * stack_tmp_stride*